### PR TITLE
Allow basic mysql support for type year - mainly for lecagy DB support

### LINKED ISSUE #7444: 3.1: "year" form field not detected as datetime part
MySQL `YEAR(4) default null` field shows up as textarea instead of dropdown year select (part of DateTime widget).
I think since the manual way works, the FormHelper's schema auto-detection should also reflect this.

Also, unknown field types should better show up as text instead of textarea IMO.

### PR DESCRIPTION
**Pull request by dereuromark**
Resolves https://github.com/cakephp/cakephp/issues/7444 for MySQL

RFC, we can either add tests and merge or close.

### CODE REVIEW

**Comment by markstory** [comment]
Seems ok to me. Needs tests though.

File: src/View/Widget/DateTimeWidget.php [hunk 1]
```diff
@@ -219,6 +219,9 @@ protected function _deconstructDate($value, $options)
             ];
         }
         try {
+            if (is_string($value) && is_numeric($value) && strlen($value) === 4) {
```
Comment (ravage84) [inline src/View/Widget/DateTimeWidget.php]: Nitpicking: It does not need to be in the try clause...

**Comment by dereuromark** [comment]
Do the tests make sense?
Does year type then also default to the datetime widget with only year present?

**Comment by dereuromark** [comment]
Hm, I don't think this is going to work out without also a Type class and some more config on top.
Since I got a basic [YearType](https://github.com/dereuromark/cakephp-shim/blob/master/src/Database/Type/YearType.php) in Shim, I will close this here.

**Pull request closed without merging**